service/matching/taskListManager.go

```diff
@@ -372,13 +376,19 @@ func (c *taskListManagerImpl) GetTaskContext(
 		WorkflowId: common.StringPtr(task.WorkflowID),
 		RunId:      common.StringPtr(task.RunID),
 	}
+
+	// reload from domainCache in case it got empty result during construction
+	if len(c.domainName) == 0 {
+		c.domainName, c.domainScope = domainNameAndMetricScope(c.domainCache, c.taskListID.domainID, c.metricsClient, metrics.MatchingTaskListMgrScope)
```

Comment:
i don't think is thread safe.  Even if it is, it will be a benign race, which will still trigger race detector errors. You will either need a lock or use an atomic type

Reply:
Sure